src/decode/xz.rs

```diff
@@ -104,11 +64,19 @@ where
             )));
         }
 
-        let footer_flags = digested.read_u16::<BigEndian>()?;
-        if flags != footer_flags {
```

Comment:
We should keep the `u16` flags abstraction, and then check that the flags are valid. Otherwise `Invalid null byte in footer` doesn't explain where this null byte comes form, which is less clear than the code before this pull request.